lib/scala/cli/src/main/scala/org/enso/cli/Command.scala

```diff
@@ -0,0 +1,102 @@
+package org.enso.cli
+
+/**
+  * Represents a top-level command in the CLI.
+  *
+  * @param name name of the command
+  * @param comment a help comment displayed in the commands help text
+  * @param related names of related command names that should redirect the user
+  *                to this command
+  * @param opts parsing logic for command's options
+  * @tparam A type returned by the command
+  */
+case class Command[A](
+  name: String,
+  comment: String,
+  related: Seq[String] = Seq()
+)(
+  val opts: Opts[A]
```

Comment:
Maybe it is worth creating an extra apply method, instead of making the default constructor curried. Just for clarity.

Reply:
Actually, this is done on purpose, to allow for syntax like:
```
val cmd = Command[Config]("name", "comment") {
   val flag1 = Opts.flag("flag", "help text.", false)
   val arg1 = Opts.positionalArgument[Int]("N")
   (flag1, arg1) mapN { (flag, arg) => (_: Config) => println(s"Arg was $arg and flag was $flag.") }
}
```

Reply:
I wanted to say that the curried constructor for a class looks weird to me. I meant something like that:

```scala
case class Foo(a: A, b: B, c: C)
object Foo {
 def apply(a: A, b: B)(c: C): Foo =
  new Foo(a, b, c)
}
```

Reply:
Ah, so I misunderstood. That's a good idea, added.